### DISCUSSION TITLE
How to do Multi-level Subcategory Table?

### DISCUSSION THREAD

**Discussion by NgYueHong** [filamentphp/filament#12353] [Help]
### Package

Table builder

### Package Version

3.2

### How can we help you?

Hi,

How can I do the multi-level "unlimited" subcategory on Filament?

This is what I have right now:

Category Table:
```
        Schema::create('categories', function (Blueprint $table) {
            $table->id();
            $table->string('name');
            $table->unsignedInteger('parent_id');
            $table->timestamps();
        });
```
Category Resource:
```
   public static function form(Form $form): Form
    {
        return $form
            ->schema([
                TextInput::make('name')
                    ->label('Category')
                    ->required()
            ]);
    }

    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                TextColumn::make('name')
                    ->label('Category'),
            ])
            ->actions([
                Tables\Actions\EditAction::make(),
                Tables\Actions\Action::make('subCategory'),
            ]);
    }
```
How can when user click on the "subCategory" button, the same table will navigate into the subcategory and allow user to add another category to that subcategory?

**Comment by D-Jordy** [comment]
you might want to do *one* of the following

- add a subcategory filter <https://filamentphp.com/docs/3.x/tables/filters/getting-started>
- use a relation manager <https://filamentphp.com/docs/3.x/panels/resources/relation-managers#creating-a-relation-manager>
- make a custom page/widget for it and dynamically set the query via the action (more advanced)
